Automattic/jetpack
# JS Components: typescriptify Dialog component

This PR rewrites the Dialog component using TypeScript

Fixes #

#### Changes proposed in this Pull Request:

* JS Components: typescriptify Dialog component

#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?



#### Testing instructions:






* Use Storybook:
```
cd projects/js-packages/storybook && pnpm run storybook:dev
```

* Take a look at
  * Dialog stories: http://localhost:50240/?path=/story/js-packages-components-dialog--interstitial-jetpack-boost
  * Protect Interstitial: http://localhost:50240/?path=/story/plugins-protect-interstitial--default
  * Protect Footer: http://localhost:50240/?path=/story/plugins-protect-footer--default
* Confirm that all the stories work as expected

_Disclaimer: there are some visual issues that we'll be addressed in follow-up PRs_

<img width="1249" alt="image" src="https://user-images.githubusercontent.com/77539/166922312-f00a0f2f-24df-41b4-b6cd-b48c9694ee72.png">
